applications/common/backend/src/main/java/pl/ds/kyanite/common/components/models/table/TableCellComponent.java

```diff
@@ -16,39 +16,135 @@
 
 package pl.ds.kyanite.common.components.models.table;
 
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
 import javax.annotation.PostConstruct;
 import javax.inject.Inject;
 import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.sling.api.resource.Resource;
-import org.apache.sling.models.annotations.Default;
 import org.apache.sling.models.annotations.DefaultInjectionStrategy;
 import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.SlingObject;
+import pl.ds.kyanite.common.components.models.table.dto.TableCellData;
+import pl.ds.kyanite.common.components.models.table.dto.TableRowData;
 
+@Slf4j
 @Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
 public class TableCellComponent {
 
   private static final String DEFAULT_TEXT = "Content";
+  private static final int DEFAULT_COLSPAN = 1;
+  private static final int DEFAULT_ROWSPAN = 1;
 
   @Inject
   @Getter
-  @Default(values = DEFAULT_TEXT)
   private String text;
 
   @Inject
-  @Getter
-  @Default(intValues = 1)
-  private int columns;
+  private Integer columns;
 
   @Inject
-  @Getter
-  @Default(intValues = 1)
-  private int rows;
+  private Integer rows;
+
+  @SlingObject
+  private Resource currentResource;
 
   @PostConstruct
   private void init() {
-    if (text.isEmpty()) {
+    if (StringUtils.isEmpty(text)) {
       text = DEFAULT_TEXT;
     }
   }
 
+  public int getColspan() {
+    return columns != null ? columns : computeColspan();
+  }
+
+  public int getRowspan() {
+    return rows != null ? rows : DEFAULT_ROWSPAN;
+  }
```

Comment:
Behavior of rows/rowspan is unchanged.

What changed is - columns are not anymore translated 1:1 to colspan. Now colspan is computed